lib/perl/Genome/Model/Tools/Vcf/EvaluateVcf.pm

```diff
@@ -0,0 +1,790 @@
+    return $stat;
+}
+
+sub stat_false_positive_exact {
+    my $self = shift;
+    my $results = $self->get_rawstats();
+    my $stat = $results->{false_positive_exact};
+    return $stat;
+}
+
+sub stat_false_positive_partial {
+    my $self = shift;
+    my $results = $self->get_rawstats();
+    my $stat = $results->{false_positive_partial};
+    return $stat;
+}
+
+sub stat_true_negatives {
+    my $self = shift;
+    my $results = $self->get_rawstats();
+    my $stat = $results->{true_negatives};
+    return $stat;
+}
+
+sub stat_exact_specificity {
+    my $self = shift;
+    my $results = $self->get_rawstats();
+
+    # exact specificity, these are not strictly accurate as the tn_bed
+    # may be significantly smaller than the target space ROI
+    
+    my $tn = $self->stat_true_negatives();
+    if ($tn == 0) {
+        return 0;
+    }
+
+    my $stat =
+      ($results->{true_negatives} - $results->{false_positive_exact}) / $tn;
+    return $stat;
+}
+
+sub stat_partial_specificity {
+    my $self    = shift;
+    my $results = $self->get_rawstats();
+
+    my $tn = $self->stat_true_negatives();
+    if ($tn == 0) {
+        return 0;
+    }
+
+    my $stat =
+      ($results->{true_negatives} - $results->{false_positive_partial}) / $tn;
+    return $stat;
+}
+
+sub stat_exact_ppv {
+    my $self    = shift;
+
+    my $fp_exact = $self->stat_false_positive_exact();
+    my $tp_exact = $self->stat_true_positive_found_exact();
+
+    my $denominator = $fp_exact + $tp_exact;
+    if ($denominator == 0) {
+        return 0;
+    }
+
+    my $stat = $tp_exact / $denominator;
+    return $stat;
+}
+
+sub stat_partial_ppv {
+    my $self    = shift;
+
+    my $fp_partial = $self->stat_false_positive_partial();
+    my $tp_partial = $self->stat_true_positive_found_partial();
+
+    my $denominator = $fp_partial + $tp_partial;
+    if ($denominator == 0) {
+        return 0;
+    }
+
+    my $stat = $tp_partial / $denominator;
+    return $stat;
+}
+
+sub stat_vcf_lines_overlapping_tn {
+    my $self    = shift;
+    my $results = $self->get_rawstats();
+    my $stat    = $results->{false_positives_in_roi};
+    return $stat;
+}
+
+sub stat_lines_specificity_in_tn_only {
+    my $self    = shift;
+    my $results = $self->get_rawstats();
+
+    # this is more accurate than the other specificity measures, but
+    # doesn't take partial into account
+
+    my $tn = $self->stat_true_negatives();
+    if ($tn == 0) {
+        return 0;
+    }
+
+    my $stat = ($tn - $results->{false_positives_in_roi}) / $tn;
+    return $stat;
+}
+
+sub help_brief {
+    return "Compare or Validate VCF files to a Gold Standard";
+}
+
+sub help_detail {
+    my $doc = q{
+
+    BACKGROUND
+    ==========
+
+    For validation, it is important to compare calls from the pipeline +
+    reporting framework and calculate sensitivity, specificity and positive
+    predictive value of the pipeline's calls versus the gold standard.
```

Comment:
The introduction has been updated now.  Let me know what you think.

On 4/8/15 9:11 PM, Travis Abbott wrote:
>
> In lib/perl/Genome/Model/Tools/Vcf/EvaluateVcf.pm 
> <https://github.com/genome/genome/pull/640#discussion_r28030753>:
>
> > +    return $stat;
> > +}
> > +
> > +sub help_brief {
> > +    return "Compare or Validate VCF files to a Gold Standard";
> > +}
> > +
> > +sub help_detail {
> > +    my $doc = q{
> > +
> > +    BACKGROUND
> > +    ==========
> > +
> > +    For validation, it is important to compare calls from the pipeline +
> > +    reporting framework and calculate sensitivity, specificity and positive
> > +    predictive value of the pipeline's calls versus the gold standard.
>
> Agreed. Also, "the gold standard" isn't particularly meaningful. This 
> is a tool for measuring similarity/difference between callsets 
> (concordance, if you prefer). If you want to call one of them "gold", 
> that's up to you. It isn't inherent in the code/algorithms involved, 
> however. Param names may have been things like "gold" when we got 
> them, and perhaps it is fine to keep them that way. That said, I want 
> to make sure that the essence of the underlying tools is understood: 
> all we are doing here is computing basic statistics to quantify the 
> similarity of two ostensibly overlapping sets (with some flexibility 
> in the definition of the relevant equivalence relation [exact match, 
> partial match]).
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/genome/genome/pull/640/files#r28030753>.
>



____
This email message is a private communication. The information transmitted, including attachments, is intended only for the person or entity to which it is addressed and may contain confidential, privileged, and/or proprietary material. Any review, duplication, retransmission, distribution, or other use of, or taking of any action in reliance upon, this information by persons or entities other than the intended recipient is unauthorized by the sender and is prohibited. If you have received this message in error, please contact the sender immediately by return email and delete the original message from all computer systems. Thank you.